core/templates/dev/head/services/UserService.js

```diff
@@ -66,6 +66,23 @@ oppia.factory('UserService', [
           data: newProfileImageDataUrl
         });
       },
+      getLoginAndLogoutUrls: function() {
+        var urlParameters = {
+          current_url: $window.location.href
+        };
+        var loginGenerationConfig = {
+          params: urlParameters
+        };
+        return $http.get('/url_handler', loginGenerationConfig).then(
```

Comment:
You don't really need `loginGenerationConfig` as a separate variable, do something like this:
```suggestion
        return $http.get('/url_handler', {params: urlParameters}).then(
```